simtools/corsika/corsika_config.py

```diff
@@ -0,0 +1,522 @@
+import logging
+import random
+
+import astropy.units as u
+from astropy.io.misc import yaml
+
+import simtools.config as cfg
+import simtools.io_handler as io
+import simtools.util.general as gen
+from simtools.util import names
+from simtools.layout.layout_array import LayoutArray
+from simtools.util.general import collectDataFromYamlOrDict
+
+__all__ = [
+    'CorsikaConfig',
+    'MissingRequiredInputInCorsikaConfigData',
+    'InvalidCorsikaInput'
+]
+
+
+class MissingRequiredInputInCorsikaConfigData(Exception):
+    pass
+
+
+class InvalidCorsikaInput(Exception):
+    pass
+
+
+class CorsikaConfig:
+    '''
+    CorsikaConfig deals with configuration for running CORSIKA. \
+    User parameters must be given by the corsikaConfigData or \
+    corsikaConfigFile arguments. An example of corsikaConfigData follows \
+    below.
+
+    .. code-block:: python
+
+    corsikaConfigData = {
+        'primary': 'proton',
+        'nshow': 10000,
+        'nrun': 1,
+        'zenith': 20 * u.deg,
+        'viewcone': 5 * u.deg,
+        'erange': [10 * u.GeV, 100 * u.TeV],
+        'eslope': -2,
+        'phi': 0 * u.deg,
+        'cscat': [10, 1500 * u.m, 0]
+    }
+
+    The remaining CORSIKA parameters can be set as a yaml file, using the argument \
+    corsikaParametersFile. When not given, corsikaParameters will be loaded \
+    from data/corsika/corsika_parameters.yml.
```

Comment:
Need to discuss in general for gamma-simtools what should be done if there are no parameters or parameter files are given. Using default values and parameter files can be very useful, but sometimes 'dangerous'.

Reply:
Yes, but we have to evaluate when it's dangerous and when it's necessary.
I don't think we should ask the user to set these corsika_parameters every time he/she wants to run an application.
The code has been designed to be flexible, meaning it's easy to change the defaults in case the user want to do so. However, it's bad for usability if we request him/her to do so every time.

Reply:
Moved into issue #124